app/src/js/AstroVectorGrid.js

```diff
@@ -56,6 +56,13 @@ export default L.AstroVectorGrid = L.VectorGrid.Protobuf.extend({
     });
     map.on(L.Draw.Event.CREATED, this.selectFeatures, this);
 
+    // somehow "this" is lost with nested binding of event handlers
+    let vectorGrid = this;
+
+    // Whenever a new poly is created, bind an event handler to it.
+    map.on("draw:created", function(e) {
```

Comment:
Instead of having two event listeners listening to the same event, maybe we can move this into the other listener on line 57. Change that line to call this.selectFeatures and then add the layer onlick listener. Or we can pass the event into selectFeatures and add the listener in there. What are your thoughts?

Reply:
I moved the event listener binding to selectFeatures.  This also resolves the other conversation.